src/nidcpower/system_tests/test_system_nidcpower.py

```diff
@@ -233,6 +233,115 @@ def test_measure_multiple(session):
         assert measurements[1].current == 0.00001
 
 
+@pytest.mark.parametrize(
+    'resource_name,channels,independent_channels,measurement_channels,expected_measured_channel',
+    [
+        ('Dev1', '1', False, None, '1'),
```

Comment:
I'm not sure how much runtime all these test cases add. But given that the bulk of the logic is dealing with channel names and that is thoroughly tested in unit tests, I think you can do a handful of test cases here and that should suffice. No need to stress things that were already stressed (redundant coverage), it's even detrimental as it can be confusing and harder to maintain.

We're more concerned with testing integration here.

This applies to the all the new tests, and perhaps the need to add more checks to the existing tests for the two modified methods.

I am not giving you specific feedback on what to add / remove, rather guideline on how to approach it.

Reply:
Thanks for the info. I have gone through the test cases once more and I think they are essential for covering the various integration scenarios, e.g. mixing different types of `resource_name` and `channels` during session initialization with different `measurement_channels` string (including passing no string, i.e. calling `session.fetch_multiple()` directly without `channels`) as well as independent vs. non-independent channels session.

I shall retain the test cases as is for now, feel free to reopen this conversation if there are any particular test cases which you think are redundant.